### PR TITLE
Update GraphQL to be closer to execution-api spec

### PR DESCRIPTION
**Pull request by shemnon**
## PR description

Update GraphQL to be closer to execution-apis spec 

Update the graphQL to be closer to the execution-api specs.
This is a departure from EIP-1767 but the core devs have made it clear
that the execution-apis repo is considered canonical, not the EIP.

Major changes:
* Add support for EIP-2930 access lists
* Add support for EIP-1559 fees
* Add transaction, block, receipt, and header raw rlp fields
* Add chainID as base level query

Some items are still fake (fee estimate, effective tip).

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Review by macfarla — COMMENTED** [review]
quick pass through - couple of minor comments. Is there notes or docs from ACD that we can link to as motivation for the move away from the spec?

> This is a departure from EIP-1767 but the core devs have made it clear
that the execution-apis repo is considered canonical, not the EIP.

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java [hunk 1]
```diff
@@ -146,6 +147,18 @@ DataFetcher<Optional<Wei>> getGasPriceDataFetcher() {
...
 
+  public DataFetcher<Optional<BigInteger>> getChainIdDataFetcher() {
+    return dataFetchingEnvironment -> {
+      final GraphQLContext graphQLContext = dataFetchingEnvironment.getGraphQlContext();
+      return graphQLContext.get(GraphQLContextType.CHAIN_ID);
+    };
+  }
+
+  public DataFetcher<Optional<Wei>> getMaxPriorityFeePerGasDataFetcher() {
+    // FIXME
```
[the remark this answers is not in the document]
Comment (shemnon) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLDataFetchers.java:158]: It needs to provide a sensible estimate.  No criteria for what sensible is.  Do we have it be node configurable?  Do we get the 25%, 50%,or 75% priority fee of the last X blocks?  etc. etc.

**Comment by shemnon** [comment]
> Is there notes or docs from ACD that we can link to as motivation for the move away from the spec?

Unfortunately no, the best I can do is a discord thread - https://discord.com/channels/595666850260713488/746566142700814426/993575909724524584

I am disheartened that the spec and EIP process is slowly breaking down.

**Review by macfarla — COMMENTED** [review]
is this a breaking change?

File: ethereum/api/src/test/resources/org/hyperledger/besu/ethereum/api/graphql/eth_getBalance_invalidAccountBlockNumber.json [hunk 2]
```diff
@@ -1,9 +1,11 @@
 {
-  "request": "{account(blockNumber:\"0x19\", address: \"0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef\") { balance } }",
```
Comment (macfarla) [inline ethereum/api/src/test/resources/org/hyperledger/besu/ethereum/api/graphql/eth_getBalance_invalidAccountBlockNumber.json:2]: does this make it a breaking change?

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLHttpServiceCorsTest.java [hunk 3]
```diff
@@ -67,7 +67,7 @@ public void requestWithNonAcceptedOriginShouldFail() throws Exception {
 
     final Request request =
         new Request.Builder()
-            .url(graphQLHttpService.url() + "/graphql?query={protocolVersion}")
```
[the remark this answers is not in the document]
Comment (shemnon) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLHttpServiceCorsTest.java:70]: protocol version is not in the new (execution-apis) spec, so we needed a test query that was in the new spec.  The query is secondary in these tests, it only needs to be a legal query.

[the remark this answers is not in the document]
Comment (macfarla) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLHttpServiceCorsTest.java:70]: got it

File: ethereum/api/src/test/resources/org/hyperledger/besu/ethereum/api/graphql/eth_getBalance_invalidAccountBlockNumber.json (hunk 2 above)
Comment (macfarla) [reply ethereum/api/src/test/resources/org/hyperledger/besu/ethereum/api/graphql/eth_getBalance_invalidAccountBlockNumber.json:2]: understood.

**Review by macfarla — APPROVED** [review]
approved. just will need the changelog moved, sorry for the delay

**Pull request merged**